doc/userguide/devguide/internals/engines/stream/applayertriggerrawreassembly.rst

```diff
@@ -0,0 +1,37 @@
+AppLayerParserTriggerRawStreamReassembly
+----------------------------------------
+
+This is a call for immediate reassembly of raw stream data by the Application Layer in case
+there is a requirement e.g. upon completion of an HTTP request.
+
+The stream engine provides the following parameters for the stream to be tracked by the different
+subsystems of Suricata. These parameters are used to track the progress of the stream in
+the respective subsystems. This serves as a communication between the different subsystems
+about how far they have already looked into the stream.
+They are all relative to the base offset of the streaming buffer region.
+
+* app layer progress
+* raw reassembly progress
+* streaming logger progress
+
+For a given streaming buffer as follows and a raw stream progress as seen by the detection
+engine
```

Comment:
stream raw progress indicates where in the stream the detection engine has already inspected, so it's a "bookmark" of sorts to keep track of where the last inspection left off and from what offset a next call should consider inspection.

Reply:
indeed. That's a great explanation. Do you think the diagram also needs changes to reflect this? Suggestions on how?

Reply:
yes, the section that is inspected by detect is *after* raw stream progress

Reply:
ah. By the `Raw Stream Progress` in the diagram i meant the `det_ctx->raw_stream_progress` but I can see how I made it confusing. I'll fix this.